spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/mail/MailSenderAutoConfiguration.java

```diff
@@ -69,9 +69,10 @@ public JavaMailSenderImpl mailSender() {
 		JavaMailSenderImpl sender = new JavaMailSenderImpl();
 		if (this.session != null) {
 			sender.setSession(this.session);
+			applyProperties(sender);
```

Comment:
On a second thought, that doesn't sound right at all that we set username/password/host when we already have a `Session` from JNDI that should give us that stuff